### PR TITLE
brightmaps: STATE works for all sprites

### PR DESCRIPTION
**Pull request by rfomin**
Fix [[doomworld]](https://www.doomworld.com/forum/topic/112333-this-is-woof-1450-apr-30-2024/?do=findComment&comment=2814151)

### CODE REVIEW

File: src/r_things.c [hunk 1]
```diff
@@ -692,6 +692,10 @@ static void R_ProjectSprite (mobj_t* thing)
     }
   vis->brightmap = R_BrightmapForSprite(thing->sprite);
 
+  const byte *tmp_brightmap = R_BrightmapForState(thing->state - states);
+  if (tmp_brightmap != nobrightmap)
+    vis->brightmap = tmp_brightmap;
```
Comment (fabiangreffrath) [inline src/r_things.c:697]: Should we move the earlier `R_BrightmapForSprite()` call into the `else` branch, so we don't have to call two brightmaps functions during each call?

Comment (rfomin) [reply src/r_things.c:696]: Done in https://github.com/fabiangreffrath/woof/pull/1750/commits/cd7147d8be6cd38fc9b9f4c606b687aaeba8dd3c

**Review by fabiangreffrath — APPROVED** [review]
Thank you!

File: src/r_things.c [hunk 2]
```diff
@@ -690,7 +690,12 @@ static void R_ProjectSprite (mobj_t* thing)
       vis->colormap[0] = spritelights[index];
       vis->colormap[1] = fullcolormap;
     }
-  vis->brightmap = R_BrightmapForSprite(thing->sprite);
+
+  const byte *tmp_brightmap = R_BrightmapForState(thing->state - states);
```
Comment (fabiangreffrath) [inline src/r_things.c:694]:
```C
vis->brightmap = R_BrightmapForState(thing->state - states);
if (vis->brightmap == nobrightmap)
  vis->brightmap = R_BrightmapForSprite(thing->sprite);
```

Comment (rfomin) [reply src/r_things.c:694]: OK, I will do it in master.

**Pull request merged**